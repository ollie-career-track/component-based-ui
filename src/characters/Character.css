.Character {
  background-color: lightpink;
  width: 300px; 

  img {
    width: 250px;
  }

  p{
    background-color: lightyellow;
  }
}