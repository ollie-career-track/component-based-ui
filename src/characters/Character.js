import React from 'react';
import PropTypes from 'prop-types';
import styles from './Character.css';

const Character = ({ name, species, img }) => (
  <div className={styles.Character}>
    <img src={img}/>
    <p>Name: {name}</p>
    <p>Species: {species}</p>
  </div>
);

Character.propTypes = {
  name: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired
};

export default Character;
