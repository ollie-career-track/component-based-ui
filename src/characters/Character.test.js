import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders Character', () => {
    const wrapper = shallow(<Character name=" Evil Morty" species="Human" img="./assets/evil-morty.png" />);
    expect(wrapper).toMatchSnapshot();
  });
});
