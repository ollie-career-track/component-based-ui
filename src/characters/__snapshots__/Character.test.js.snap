// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<div
  className="Character"
>
  <img
    src="./assets/evil-morty.png"
  />
  <p>
    Name: 
     Evil Morty
  </p>
  <p>
    Species: 
    Human
  </p>
</div>
`;
